### PR TITLE
Default values

### PR DESCRIPTION
**Pull request by tung-vu-td**
### What
Implementation of Default values feature for td-ios-sdk. Default value is value that will be automatically added to an event when that event is added to the specified table and database of that default value.

With this feature, SDK users can set default values for events targeting:
- Any tables
- Tables with specific name
- All tables in database with a specific name
- Specific table in a specific database

### Why
- SDK users only need to write one method call to add default values to events, compared to previously they would have to remember to add that value whenever adding new events.
- Default values also affect custom events which previously not possible for SDK users to add new customized column. This will allow SDK user to implement feature such as tracking user session themself.

### CODE REVIEW

**Comment by huylenq** [comment]
Had some naming/signature comments over https://github.com/treasure-data/td-android-sdk/pull/52. They're also relevant here.

File: TreasureData/TreasureData.h [hunk 1]
```diff
@@ -455,6 +455,38 @@ typedef void (^ErrorHandler)(NSString* _Nonnull errorCode, NSString* _Nullable e
...
+/**
+ * Set default `value` for  `key` in all new events targeting `database` and `table`.
+ * When `database` and/or `table` parameters are nil, the nil parameter acts like a wild card that allows to set specified key value pair to new events added to any database (if `database` is nil) and/or to any table (if `table` is nil).
+ * For example, if you pass nil to both `database` and `table` parameters, all new events will have specified default value.
+ * @param value default value for `key`
+ * @param key the event's key that default value is set to, corresponding to column in table.
+ * @param database the database to set default value to. If nil, specified table of any database will have new events with the added default value.
+ * @param table the table to set default value to. If nil, any table of specified database will have new events with the added default value.
+ */
+- (void)setDefaultValue:(nonnull id)value forKey:(nonnull NSString *)key database:(nullable NSString *)database table:(nullable NSString *)table;
```
Comment (huylenq) [inline TreasureData/TreasureData.h:469]: Provide the same method variants as `addEvent` instead of nullable arguments?

Comment (tung-vu-td) [reply TreasureData/TreasureData.h:469]: Similar to comment in td-android-sdk PR

**Review by huylenq — APPROVED** [review]
LGTM

**Review by nukala — APPROVED** [review]
Reviewed for Data Exfiltrations

**Pull request merged**